sdl-core/src/main/scala/io/smartdatalake/config/ConfigLoader.scala

```diff
@@ -91,10 +91,22 @@ object ConfigLoader extends SmartDataLakeLogger {
     val hadoopConf: Configuration = new Configuration() // note that we could not yet load additional hadoop/spark configurations set in the configuration files
 
     // Search locations for config files
-    val configFiles = hadoopPaths.flatMap(
-      location => if (ClasspathConfigFile.canHandleScheme(location)) Seq(ClasspathConfigFile(location))
-      else getFilesInBfsOrder(location)(location.getFileSystem(hadoopConf))
+    val configFiles = hadoopPaths.flatMap( location =>
+        if (ClasspathConfigFile.canHandleScheme(location))
+          Seq(ClasspathConfigFile(location))
+        else
+          try {
+            var fs = location.getFileSystem(hadoopConf)
+            if(fs.getFileStatus(location).isDirectory & !RemoteIteratorWrapper(fs.listFiles(location, true)).exists(path =>
+                configFileExtensions.contains(path.getPath.toUri.toString.split("\\.").last)
+              ))
+                throw ConfigurationException(s"The provided directory path ${location.toUri.getPath} does not contain any valid config files")
+            getFilesInBfsOrder(location)(location.getFileSystem(hadoopConf))
```

Comment:
```suggestion
            val files = getFilesInBfsOrder(location)(location.getFileSystem(hadoopConf))
            if (files.isEmpty){
              throw ConfigurationException(s"The provided directory path ${location.toUri.getPath} does not contain any valid config files")
            }
            files
```

Reply:
Yes, looks very nice and much shorter. :)